lang/elaborator/src/normalizer/eval.rs

```diff
@@ -95,16 +99,16 @@ impl Eval for DotCall {
                     }
                 }
             }
-            Val::Comatch { body, .. } => beta_comatch(prg, body, name, &args),
-            Val::Neu { exp } => Ok(Rc::new(Val::Neu {
-                exp: Neu::DotCall {
-                    span: *span,
-                    kind: *kind,
-                    exp: Rc::new(exp),
-                    name: name.to_owned(),
-                    args,
-                },
-            })),
+            Val::LocalComatch(val::LocalComatch { body, .. }) => {
+                beta_comatch(prg, body, name, &args)
+            }
+            Val::Neu(exp) => Ok(Rc::new(Val::Neu(Neu::DotCall(val::DotCall {
```

Comment:
Thanks! I did this in [b3a6ef5](https://github.com/polarity-lang/polarity/pull/197/commits/b3a6ef52e21edb7ddfe1b34e2c68a0f8c5cf49fe) for the changes I found in `eval.rs`.

Reply:
Thinking about it, it could probably be simplified even more, by embedding directly into `Rc<Val>`. I.e. `Ok(DotCall { ... }.into())` if type inference allows for it. Not sure if that hides to much of the structure, though.